x-hooks/use-breakpoint.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+
+import { BOOTSTRAP_BREAKPOINTS } from "helpers/constants"
+
+import useDebouncedCallback from "x-hooks/use-debounced-callback";
+
+const getCurrentBreakPoint = (currentWidth: number) => {
+  if (currentWidth < BOOTSTRAP_BREAKPOINTS.sm) return "xs";
+  else if (currentWidth < BOOTSTRAP_BREAKPOINTS.md) return "sm";
+  else if (currentWidth < BOOTSTRAP_BREAKPOINTS.lg) return "md";
+  else if (currentWidth < BOOTSTRAP_BREAKPOINTS.xl) return "lg";
+  else if (currentWidth < BOOTSTRAP_BREAKPOINTS.xxl) return "xl";
+
+  return "xxl";
+}
+
+const isMobileUserAgent = () => {
+  if (!window) return false;
+
+  const userAgentMatch = agent => !!navigator.userAgent.match(agent);
+
+  if (userAgentMatch(/Android/i) ||
```

Comment:
this can be transformed into a `[].some()`:

```ts

const mobileUserAgents = [ /webOS/i, /* ... */ ]
const userAgentMatch = agent => !!navigator.userAgent.match(agent);

return mobileUserAgents.some(userAgentMatch);
```

`some` will break out on the first true (and return true) or false if none is true